zyte_common_items/processors.py

```diff
@@ -49,3 +49,22 @@ def _from_zp_breadcrumb(value: zp_Breadcrumb) -> Breadcrumb:
         else:
             results.append(item)
     return results
+
+
+def brand_processor(value: Any) -> Any:
+    """Convert the data into a brand name if possible.
+
+    Supported inputs are :class:`~parsel.selector.Selector`,
+    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
+    Other inputs are returned as is.
+    """
+
+    if isinstance(value, SelectorList):
+        if len(value) == 0:
+            return None
+        value = value[0]
+
+    if isinstance(value, (Selector, HtmlElement)):
+        return extract_brand_name(value, search_depth=2)
```

Comment:
This seems to be a pattern which is going to be common for all processors. I'm fine with merging it as-is, when we only have 2 processors, but once we add 3rd, we should refactor it; no harm in refactoring it now as well.

Reply:
Yeah.